misc/Kconfig

```diff
@@ -317,8 +317,15 @@ config BOOTLOADER_MCUBOOT
 	help
 	  This option signifies that the target uses MCUboot as a bootloader,
 	  or in other words that the image is to be chain-loaded by MCUboot.
-	  This sets several required options in order for the image generated
-	  to be bootable using the MCUboot open source bootloader.
+	  This sets several required build system and Device Tree options in
+	  order for the image generated to be bootable using the MCUboot open
+	  source bootloader. Currently this includes:
+	  - Setting TEXT_LOAD_OFFSET to a default value that allows space for
+	  the MCUboot image header
+	  - Activating SW_VECTOR_RELAY on Cortex-M0 targets with no built-in
+	  vector relocation mechanisms
+	  - Including subsys/dfu/boot/mcuboot.overlay when building the Device
+	  Tree in order to place and link the image at the slot0 offset
```

Comment:
PR overall looks good, just some nits:

- The alignment in these help bullet list items looks off
- Do we really want to start putting DTS files under subsys/dfu/, instead of under dts/?

I know we have per-board dts files in the board directories, but that is because we want those to be able to be self-contained eventually. There is already a dts/common/ directory, why not use that? (Honest question, I'm open to either solution).

Reply:
>The alignment in these help bullet list items looks off

See comment below.

>Do we really want to start putting DTS files under subsys/dfu/, instead of under dts/?

I really don't mind, I like your suggestion so I'll do that.

Reply:
Actually on second thought, I think the reason I placed it outside of `dts/` is because in there there are only `.dtsi` files and not `.dts`.

Reply:
^^

@galak do you care if dts overlay files go into dts/common alongside the dtsi files? Or is that directory only for dtsi?